### ISSUE TITLE
ecdh_curves P-224 causing localhost failed to run in CentOS 7 docker environment

### ISSUE THREAD

**Issue by k-ryo** [socketry/localhost#7]
## Description

I'm running an app using CentOS 7 docker image and found that CentOS 7's own OpenSSL has no P-224 support hence causing localhost to fail. ( And most likely all OpenSSL built without enable-ec_nistp_64_gcc_128 option )

```
$ openssl ecparam -list_curves
  secp256k1 : SECG curve over a 256 bit prime field
  secp384r1 : NIST/SECG curve over a 384 bit prime field
  secp521r1 : NIST/SECG curve over a 521 bit prime field
  prime256v1: X9.62/SECG curve over a 256 bit prime field
```

Removing P-224 from lib/localhost/authority.rb:129 solved the issue.

## Error trace

```
$ falcon serve -b https://0.0.0.0:3000 --hybrid
ruby/2.6.3 isn't supported by this pry-doc version

        Called from: /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/actionpack-5.2.2.1/lib/action_dispatch/middleware/stack.rb:37:in `new'.
 4.67s: <Async::Task:0x2ae30a0eaf20 failed>
      |   OpenSSL::SSL::SSLError: unknown group
      |   → /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/localhost-1.1.5/lib/localhost/authority.rb:129 in `ecdh_curves='
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/localhost-1.1.5/lib/localhost/authority.rb:129 in `block in server_context'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/localhost-1.1.5/lib/localhost/authority.rb:118 in `tap'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/localhost-1.1.5/lib/localhost/authority.rb:118 in `server_context'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/falcon-0.31.0/lib/falcon/endpoint.rb:33 in `build_ssl_context'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/falcon-0.31.0/lib/falcon/endpoint.rb:27 in `ssl_context'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-http-0.41.2/lib/async/http/endpoint.rb:168 in `build_endpoint'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-http-0.41.2/lib/async/http/endpoint.rb:178 in `endpoint'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-http-0.41.2/lib/async/http/endpoint.rb:192 in `each'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-io-1.23.1/lib/async/io/endpoint.rb:100 in `bound'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-io-1.23.1/lib/async/io/shared_endpoint.rb:29 in `bound'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/falcon-0.31.0/lib/falcon/command/serve.rb:111 in `block in run'
      |     /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-1.17.1/lib/async/task.rb:204 in `block in make_fiber'
Traceback (most recent call last):
	12: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-1.17.1/lib/async/task.rb:204:in `block in make_fiber'
	11: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/falcon-0.31.0/lib/falcon/command/serve.rb:111:in `block in run'
	10: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-io-1.23.1/lib/async/io/shared_endpoint.rb:29:in `bound'
	 9: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-io-1.23.1/lib/async/io/endpoint.rb:100:in `bound'
	 8: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-http-0.41.2/lib/async/http/endpoint.rb:192:in `each'
	 7: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-http-0.41.2/lib/async/http/endpoint.rb:178:in `endpoint'
	 6: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/async-http-0.41.2/lib/async/http/endpoint.rb:168:in `build_endpoint'
	 5: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/falcon-0.31.0/lib/falcon/endpoint.rb:27:in `ssl_context'
	 4: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/falcon-0.31.0/lib/falcon/endpoint.rb:33:in `build_ssl_context'
	 3: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/localhost-1.1.5/lib/localhost/authority.rb:118:in `server_context'
	 2: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/localhost-1.1.5/lib/localhost/authority.rb:118:in `tap'
	 1: from /.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/localhost-1.1.5/lib/localhost/authority.rb:129:in `block in server_context'
/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/localhost-1.1.5/lib/localhost/authority.rb:129:in `ecdh_curves=': unknown group (OpenSSL::SSL::SSLError)
```

**Comment by ioquatix** [comment]
I don't know a lot about the implications of this, but do you think you can make a PR and explain how this affects security?

Of course it's only for self-signed certificates... so maybe it doesn't matter :)

**Comment by k-ryo** [comment]
I would not imagine this will affects security but again I'm no crypto expert.
The issue here is that default OpenSSL on CentOS does not have P-224 built in so localhost will failed to run, that's the only point I'm tackling basically. :)

Also, with the present of P-256, P-384, P-521 ( the more common ones ), personally I do not see any risk by taking P-224 off.

**Comment by ioquatix** [comment]
Okay can you make a PR?

**Comment by ioquatix** [comment]
Released in `v1.1.6`.
